CMakeLists.txt

```diff
@@ -21,50 +18,51 @@ endif()
 #####################################
 # Define Options
 #####################################
-option(BUILD_EXAMPLES "Builds the examples in the project" ON)
-option(BUILD_TESTING  "Builds the tests in the project"    ON)
+option(CROW_BUILD_EXAMPLES "Build the examples in the project" ON )
+option(CROW_BUILD_TESTS    "Build the tests in the project"    ON )
+option(CROW_AMALGAMATE     "Combine all headers into one"      OFF)
 
 #####################################
-# Define CMake Module Imports
+# Define Targets
 #####################################
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/dependencies.cmake)
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/compiler_options.cmake)
+add_library(crow INTERFACE)
+target_include_directories(crow INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
-#####################################
-# Define project-wide imports
-#####################################
-# this can be alternatively (and as recommended way) done with target_include_directories()
-if(BUILD_EXAMPLES OR BUILD_TESTING)
-	set(PROJECT_INCLUDE_DIR
-		${CMAKE_CURRENT_SOURCE_DIR}/include
-	)
+find_package(Boost COMPONENTS date_time REQUIRED)
+find_package(ZLIB REQUIRED)
```

Comment:
ZLIB would only be needed for `unittest.cpp` and `example_compression.cpp`. other than that it should be optional similar to OpenSSL

Reply:
That is correct, will be taken care of. 👍